### DISCUSSION TITLE
Issues running Theengs Explorer

### DISCUSSION THREAD

**Discussion by DigiH** [theengs/explorer#3] [Q&A]
Hi @koenvervloesem,

hopefully you can give me a hint as to what my problem might be whenever I'm trying to start Explorer.

All I'm getting is perpetually looping

```
Task exception was never retrieved
future: <Task finished name='Task-1458' coro=<TheengsExplorerApp.detection_callback() done, defined at __init__.py:36> exception=LookupError(<ContextVar name='active_app' at 0x113b61130>)>
Traceback (most recent call last):
  File "__init__.py", line 39, in detection_callback
    await self.scroll_view.update(self.device_table.render(), home=False)
  File "/Users/me/.pyenv/versions/3.8.5/lib/python3.8/site-packages/textual/widgets/_scroll_view.py", line 89, in update
    await self.window.update(renderable)
  File "/Users/me/.pyenv/versions/3.8.5/lib/python3.8/site-packages/textual/views/_window_view.py", line 39, in update
    layout.add(self.widget)
  File "/Users/me/.pyenv/versions/3.8.5/lib/python3.8/site-packages/textual/layouts/vertical.py", line 28, in add
    self._max_widget_width = max(widget.app.measure(widget), self._max_widget_width)
  File "/Users/me/.pyenv/versions/3.8.5/lib/python3.8/site-packages/textual/message_pump.py", line 59, in app
    return active_app.get()
LookupError: <ContextVar name='active_app' at 0x113b61130>
```

Luckily I finally managed to get this to paste here, as otherwise stopping the process always cleared it all.

Any idea?

Thanks

**Comment by koenvervloesem** [comment]
I've never seen this error. What commit is the code you are running? I've pushed a new commit 15 minutes ago, and this changed something about the scanning.

Other questions:

- Which operating system?
- Which versions of Bleak, TheengsDecoder and Textual?

**Reply by DigiH** [reply]
> I've never seen this error. What commit is the code you are running? I've pushed a new commit 15 minutes ago, and this changed something about the scanning.

Made sure I had all your latest commits

> * Which versions of Bleak, TheengsDecoder and Textual?

Also all the latest, 0.14.3, development, so 0.4.0 plus new commits, 0.1.18

> * Which operating system?

Now probably coming to the culprit, with macOS 10.15.7

**Reply by koenvervloesem** [reply]
Mmmm, I haven't used Bleak on macOS yet, but I've seen quite some 'difficult cases' in Bleak's issues...

Did it also occur with previous commits, such as https://github.com/theengs/explorer/commit/b84b36e6cbced6dc13729235df8a1c2949709a9c?

**Reply by DigiH** [reply]
Yes, the same with previous versions :(

From the Textual examples directory I can run everything fine.

Also updated my Python, even with Textual saying  `Python 3.7 or above` is fine.

**Reply by DigiH** [reply]
Let me try a more up to date version of macOS tomorrow. Have had a few different behaviours with my current version and later versions recently.

**Reply by koenvervloesem** [reply]
I'm not sure whether the problem is caused by Bleak or Textual, but maybe you can find some troubleshooting tips for Bleak here: https://bleak.readthedocs.io/en/latest/troubleshooting.html#macos-bugs

**Reply by DigiH** [reply]
One thing obvious with bleak, or any other BT implementation on macOS/iOS is the replacement of MAC addresses by uuids, as described in my initial experience with Theengs Gateway and which you commented with the Bleak issue

https://github.com/theengs/gateway/issues/12

So that shouldn't be the problem here, and Gateway is still running fine with the obvious quirks.

I'll also have a look at your link, to see if any other macOS Bleak issues might be the root of my problem.

With Textual I'm more confident, as all of Textualize's visuals, screenshots and videos, are all macOS based, even if I couldn't find any OS version info so far.

Thanks

**Reply by DigiH** [reply]
Had a look at the other suggested solutions with Bleak from your link, but none of them made any difference to the above error loop, and with Gateway using Bleak working fine in Terminal as well I'm thinking it might be on the Textual side, especially with the

`LookupError: <ContextVar name='active_app' at 0x113b61130>`

Should probably test on a later macOS version, but might wait a bit, if possibly any other later macOS user replies, before I dive into going through all the Brew, Python3 and packages install from a slow external drive ;)

**Reply by koenvervloesem** [reply]
Mmmm, maybe open an issue in the Textual repository? At least they should be able to give you some troubleshooting tips.

**Reply by DigiH** [reply]
That's what I thought as well, but wanted to wait to see if you're happy enough with it at the current stage of Explorer. You'll also be better suited if they ask particulars about Explorer.

**Reply by koenvervloesem** [reply]
Sure, go ahead, I'll follow along (I'm watching the repository anyway). As explained in https://github.com/theengs/explorer/discussions/4 I want to make sure Theengs Explorer works on the three major operating systems before publishing a package. I'm only using Linux myself.

Make sure to mention the Textual version, as they're actively working on the css branch now and the 0.1.x branch is in maintenance mode (but the only officially released one for now).

**Comment by DigiH** [comment]
Just saw this in the textual issues:

https://github.com/Textualize/textual/issues/127

**Reply by koenvervloesem** [reply]
Well, that was the warning I fixed with the latest commit :-)

**Reply by DigiH** [reply]
Sorry, was just about to delete again, as I copied and deleted it in the above thread when I realised that, always catches me out here in the discussions ;)

**Comment by koenvervloesem** [comment]
Hi @DigiH have you already found the time to try Explorer in a more recent macOS version?

**Reply by DigiH** [reply]
Not yet, as I want to wait to see what others report back with their already installed/running OS versions.
